### ISSUE TITLE
Ubuntu 20.04 VM crashes randomly on new MacBook Pros

### ISSUE THREAD

**Issue by frank7y** [utmapp/UTM#3370]
**Describe the issue**
The Ubuntu 20.04 VM created following the website instructions randomly crashes during usage. It doesn't happen really often, but still is bad. Like one time every hour or two maybe. I've tried changing the display from virtio-ramfb to virtio-ramfb-gl with no success. If it can be of some help I'm using the VM with VSCode on macOS connected to the VM via SSH, but the cpu usage on the vm is not heavy at all.

**Configuration**
* UTM Version:  2.3.1
* OS Version: macOS Monterey 12.0.1
* Apple Silicon (MBP 14")

**Crash log**
These are the crash logs for three of the crashes.
[CrashLogs.zip](https://github.com/utmapp/UTM/files/7656421/CrashLogs.zip)

**Debug log**
This is the debug log after the last of the three crashes had happened.
[debug.log](https://github.com/utmapp/UTM/files/7656420/debug.log)

**Upload VM**
[config.plist.zip](https://github.com/utmapp/UTM/files/7656418/config.plist.zip)

**Comment by pawelwiejkut** [comment]
Hi,

I'm just a regular user - but from my experience, I would suggest changing the display card to no gl if you don't need this:
![](https://pawelwiejkut.dev/abap_1909_dev_m1/utm4a.png)

The second advice is to turn off power nap:
![](https://pawelwiejkut.dev/abap_1909_dev_m1/utm4a.png)

With these settings, my Ubuntu machine is not crashing anymore. Full configuration can be found here:
https://pawelwiejkut.dev/posts/abap1909_dev_m1/

**Comment by conath** [comment]
Some technical analysis: debug log contains no errors. This VM is running in ARM hypervisor mode. macOS is current public release. **UTM version is outdated.**

@frank7y Multiple bug fixes are included in UTM 2.4.0 and 2.4.1. Please update to UTM 2.4.1 and re-open this issue if the crash still occurs.

<details>
  <summary>Crash log excerpts</summary>

Crash log file `UTM-2021-12-05-102913.ips`:

```log
Process:               UTM [40261]
Path:                  /Applications/UTM.app/Contents/MacOS/UTM
Identifier:            com.utmapp.UTM
Version:               2.3.1 (38)
Code Type:             ARM-64 (Native)
Parent Process:        launchd [1]
User ID:               501

Date/Time:             2021-12-05 10:29:10.5207 +0100
OS Version:            macOS 12.0.1 (21A559)
Report Version:        12
Anonymous UUID:        55F37BC5-7C4B-7CA8-27BA-CA6F77C65622

Sleep/Wake UUID:       61947283-D4AA-48BF-8D6F-50D850F7EF8E

Time Awake Since Boot: 220000 seconds
Time Since Wake:       3896 seconds

System Integrity Protection: enabled

Crashed Thread:        3

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000010
Exception Codes:       0x0000000000000001, 0x0000000000000010
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    Namespace SIGNAL, Code 11 Segmentation fault: 11
Terminating Process:   exc handler [40261]

VM Region Info: 0x10 is not in any region.  Bytes before following region: 4339498992
      REGION TYPE                    START - END         [ VSIZE] PRT/MAX SHRMOD  REGION DETAIL
      UNUSED SPACE AT START
--->  
      __TEXT                      102a78000-102d8c000    [ 3152K] r-x/r-x SM=COW  ...nts/MacOS/UTM

[…]

Thread 3 Crashed:
0   libobjc.A.dylib               	       0x1bde5f444 AutoreleasePoolPage::releaseUntil(objc_object**) + 144
1   libobjc.A.dylib               	       0x1bde5f47c AutoreleasePoolPage::releaseUntil(objc_object**) + 200
2   libobjc.A.dylib               	       0x1bde5be18 objc_autoreleasePoolPop + 208
3   CoreFoundation                	       0x1be055f58 _CFAutoreleasePoolPop + 32
4   CoreFoundation                	       0x1be0b33c8 __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__ + 32
5   CoreFoundation                	       0x1be0b2fc4 __CFRunLoopDoTimer + 1076
6   CoreFoundation                	       0x1be0b2a50 __CFRunLoopDoTimers + 392
7   CoreFoundation                	       0x1be0975cc __CFRunLoopRun + 1944
8   CoreFoundation                	       0x1be096694 CFRunLoopRunSpecific + 600
9   HIToolbox                     	       0x1c6c0c990 RunCurrentEventLoopInMode + 292
10  HIToolbox                     	       0x1c6c0c704 ReceiveNextEventCommon + 552
11  HIToolbox                     	       0x1c6c0c4c4 _BlockUntilNextEventMatchingListInModeWithFilter + 72
12  AppKit                        	       0x1c0bebaec _DPSNextEvent + 844
13  AppKit                        	       0x1c0bea390 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1332
14  AppKit                        	       0x1c0bdc3b0 -[NSApplication run] + 596
15  AppKit                        	       0x1c0bada84 NSApplicationMain + 1064
16  SwiftUI                       	       0x1e29891f0 specialized runApp(_:) + 148
17  SwiftUI                       	       0x1e34ab1b4 runApp<A>(_:) + 260
18  SwiftUI                       	       0x1e2f3e654 static App.main() + 128
19  UTM                           	       0x102af950c 0x102a78000 + 529676
20  UTM                           	       0x102af773c 0x102a78000 + 522044
21  dyld                          	       0x1032150f4 start + 520
```

Crash log file `UTM-2021-12-05-110238.ips`:

```logDate/Time:             2021-12-05 11:02:37.9192 +0100
OS Version:            macOS 12.0.1 (21A559)
Report Version:        12
Anonymous UUID:        55F37BC5-7C4B-7CA8-27BA-CA6F77C65622

Sleep/Wake UUID:       61947283-D4AA-48BF-8D6F-50D850F7EF8E

Time Awake Since Boot: 220000 seconds
Time Since Wake:       5904 seconds

System Integrity Protection: enabled

Crashed Thread:        4

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000010
Exception Codes:       0x0000000000000001, 0x0000000000000010
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    Namespace SIGNAL, Code 11 Segmentation fault: 11
Terminating Process:   exc handler [40630]

VM Region Info: 0x10 is not in any region.  Bytes before following region: 4373970928
      REGION TYPE                    START - END         [ VSIZE] PRT/MAX SHRMOD  REGION DETAIL
      UNUSED SPACE AT START
--->  
      __TEXT                      104b58000-104e6c000    [ 3152K] r-x/r-x SM=COW  ...nts/MacOS/UTM

[…]

Thread 4 Crashed:
0   libobjc.A.dylib               	       0x1bde5f444 AutoreleasePoolPage::releaseUntil(objc_object**) + 144
1   libobjc.A.dylib               	       0x1bde5f47c AutoreleasePoolPage::releaseUntil(objc_object**) + 200
2   libobjc.A.dylib               	       0x1bde5be18 objc_autoreleasePoolPop + 208
3   CoreFoundation                	       0x1be055f58 _CFAutoreleasePoolPop + 32
4   CoreFoundation                	       0x1be0b33c8 __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__ + 32
5   CoreFoundation                	       0x1be0b2fc4 __CFRunLoopDoTimer + 1076
6   CoreFoundation                	       0x1be0b2a50 __CFRunLoopDoTimers + 392
7   CoreFoundation                	       0x1be0975cc __CFRunLoopRun + 1944
8   CoreFoundation                	       0x1be096694 CFRunLoopRunSpecific + 600
9   HIToolbox                     	       0x1c6c0c990 RunCurrentEventLoopInMode + 292
10  HIToolbox                     	       0x1c6c0c704 ReceiveNextEventCommon + 552
11  HIToolbox                     	       0x1c6c0c4c4 _BlockUntilNextEventMatchingListInModeWithFilter + 72
12  AppKit                        	       0x1c0bebaec _DPSNextEvent + 844
13  AppKit                        	       0x1c0bea390 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1332
14  AppKit                        	       0x1c0bdc3b0 -[NSApplication run] + 596
15  AppKit                        	       0x1c0bada84 NSApplicationMain + 1064
16  SwiftUI                       	       0x1e29891f0 specialized runApp(_:) + 148
17  SwiftUI                       	       0x1e34ab1b4 runApp<A>(_:) + 260
18  SwiftUI                       	       0x1e2f3e654 static App.main() + 128
19  UTM                           	       0x104bd950c 0x104b58000 + 529676
20  UTM                           	       0x104bd773c 0x104b58000 + 522044
21  dyld                          	       0x1050110f4 start + 520
```

The third file is seemingly incomplete, but has a very similar crash.
</details>

**Comment by osy** [comment]
Seems to be #3515
